### PR TITLE
[storage] DB::open_as_secondary()

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

So we can run read only tools while the node is still running.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?
Y
## Test Plan

unit test.

### CODE REVIEW

File: storage/libradb/src/lib.rs [hunk 1]
```diff
@@ -109,12 +109,8 @@ pub struct LibraDB {
 }
 
 impl LibraDB {
-    pub fn open<P: AsRef<Path> + Clone>(
-        db_root_path: P,
-        readonly: bool,
-        prune_window: Option<u64>,
-    ) -> Result<Self> {
-        let column_families = vec![
+    fn column_families() -> Vec<ColumnFamilyName> {
```
Comment (zekun000) [inline storage/libradb/src/lib.rs:112]: const fn?

File: storage/libradb/src/lib.rs [hunk 2]
```diff
@@ -127,32 +123,67 @@ impl LibraDB {
...
+            state_store: Arc::new(StateStore::new(Arc::clone(&db))),
+            transaction_store: Arc::new(TransactionStore::new(Arc::clone(&db))),
+            system_store: SystemStore::new(Arc::clone(&db)),
+            pruner: prune_window.map(|n| Pruner::new(Arc::clone(&db), n)),
+        }
+    }
+
+    pub fn open<P: AsRef<Path> + Clone>(
+        db_root_path: P,
+        readonly: bool,
```
Comment (thefallentree) [inline storage/libradb/src/lib.rs:145]: should we make readonly always use "open as secondary ?"

File: storage/libradb/src/lib.rs (hunk 1 above)
Comment (msmouse) [reply storage/libradb/src/lib.rs:112]: It's allocating a `Vec` on the heap.. so can't be const..

File: storage/libradb/src/lib.rs (hunk 2 above)
Comment (msmouse) [reply storage/libradb/src/lib.rs:145]: We'd need to update call sites to pass in the secondary path. (Using a tempdir is not always the right thing to do.)

**Comment by msmouse** [comment]
@zekun000  走好不送

/land

**Pull request closed without merging**